### PR TITLE
snap: call fsync before close db file

### PR DESCRIPTION
**Pull request by xiang90**
We need to ensure the completeness of the saved db file.

/cc @heyitsanthony

### CODE REVIEW

File: snap/db.go [hunk 1]
```diff
@@ -32,6 +32,9 @@ func (s *Snapshotter) SaveDBFrom(r io.Reader, id uint64) error {
 		return err
 	}
 	_, err = io.Copy(f, r)
+	if err == nil {
+		err = f.Sync()
+	}
 	f.Close()
```
Comment (jonboulle) [inline snap/db.go]: ignored error here..

Comment (heyitsanthony) [inline snap/db.go]: the error is handled after the `f.Close()`, so it should still be OK

**Comment by heyitsanthony** [comment]
lgtm

File: snap/db.go (hunk 1 above)
Comment (xiang90) [inline snap/db.go]: We should also handle the error returned by f.Close(). But it is another issue. I will send another pull request to fix it.

**Pull request merged**